app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt

```diff
@@ -0,0 +1,152 @@
+package com.mailerdaemon.app.clubs
+
+import android.annotation.SuppressLint
+import android.os.Build
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.MotionEvent
+import android.view.MotionEvent.ACTION_BUTTON_PRESS
+import android.view.View
+import android.view.ViewGroup
+import androidx.swiperefreshlayout.widget.CircularProgressDrawable
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import com.mailerdaemon.app.BuildConfig
+import com.mailerdaemon.app.R
+import com.mailerdaemon.app.utils.ChromeTab
+import kotlinx.android.synthetic.main.fragment_club_detail.view.club_icon
+import kotlinx.android.synthetic.main.fragment_club_detail.view.club_des
+import kotlinx.android.synthetic.main.fragment_club_detail.view.club_fb
+import kotlinx.android.synthetic.main.fragment_club_detail.view.club_insta
+import kotlinx.android.synthetic.main.fragment_club_detail.view.club_members
+import kotlinx.android.synthetic.main.fragment_club_detail.view.club_name
+import kotlinx.android.synthetic.main.fragment_club_detail.view.club_web
+import kotlinx.android.synthetic.main.fragment_club_detail.view.club_youtube
+import kotlinx.android.synthetic.main.fragment_club_detail.view.web
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.IOException
+import java.nio.charset.StandardCharsets
+
+@Suppress("DEPRECATION")
+class ClubDetailBottomSheet : BottomSheetDialogFragment() {
+
+    private var i = 0
+    private var it = 0
+    private var chromeTab: ChromeTab? = null
+
+    @SuppressLint("ClickableViewAccessibility")
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val view = inflater.inflate(R.layout.fragment_club_detail, container, false)
+        chromeTab = ChromeTab(context)
+        view.web.isHorizontalScrollBarEnabled = true
+        view.club_icon.hierarchy.setProgressBarImage(CircularProgressDrawable((context)!!))
+        if (BuildConfig.DEBUG && arguments == null) {
+            error("Assertion failed")
+        }
+        i = requireArguments().getInt("club_id")
+        getJson(view)
+        return view
+    }
+
+    private fun getJson(view: View) {
+        val json: String
+        try {
+            val inputStream = (context)!!.assets.open("clubs.json")
+            val size = inputStream.available()
+            val buffer = ByteArray(size)
+            inputStream.read(buffer)
+            inputStream.close()
+            json = String(buffer, StandardCharsets.UTF_8)
+            val jsonObject = JSONObject(json)
```

Comment:
All the above things can be done in a better way please see it's already used in contact activity

Reply:
I was earlier trying the same thing as in club activity sir but it was giving errors when converted to kotlin
so I found this easy